src/components/Reports/DesignationAccountsReport/DesignationAccountsReport.tsx

```diff
@@ -39,21 +43,60 @@ export const DesignationAccountsReport: React.FC<Props> = ({
     setActiveDesignationAccount,
   ] = useSetActiveDesignationAccountMutation();
 
-  const handleCheckToggle = async (
+  const handleCheckToggle = (
     event: React.ChangeEvent<HTMLInputElement>,
     designationAccountId: string,
   ) => {
-    await setActiveDesignationAccount({
+    const optimisticResponse = true;
+    setActiveDesignationAccount({
       variables: {
         input: {
           accountListId,
           active: event.target.checked,
           designationAccountId,
         },
       },
-    });
+      optimisticResponse: {
+        setActiveDesignationAccount: {
+          id: designationAccountId,
+          __typename: 'SetActiveDesignationAccountResponse',
+          active: event.target.checked,
+        },
+      },
+      update: (cache) => {
+        if (!optimisticResponse) return;
+
+        const query = {
+          query: DesignationAccountsDocument,
+          variables: {
+            accountListId,
+          },
+        };
 
-    return refetch();
+        const dataFromCache = cache.readQuery<DesignationAccountsQuery>(query);
+
+        const data = {
+          designationAccounts: dataFromCache?.designationAccounts.map(
+            (designationAccountsGroup) => ({
+              ...designationAccountsGroup,
+              designationAccounts: designationAccountsGroup.designationAccounts.map(
+                (designationAccount) => {
+                  return designationAccount.id === designationAccountId
+                    ? {
+                        ...designationAccount,
+                        id: designationAccountId,
+                        active: event.target.checked,
```

Comment:
I think this would be better to get from the mutation response than the client event in case the server doesn't save it or gets 2 updates at once or something. Unlikely but the server should be the source of truth.